bin/generate.js

```diff
@@ -1,43 +1,51 @@
-'use strict';
+#!/usr/bin/env node
 const Handlebars = require('handlebars');
 const fs = require('fs');
 const path = require('path');
-const config = require(__dirname + '/../config.json');
-const template = fs.readFileSync(__dirname + '/../template.handlebars',  'utf-8');
-
-let pages = path.join(__dirname, '..', 'pages');
-if(config.folder) {
-  pages = path.join(__dirname, '..', config.folder);
-}
-
-let outputDir = path.join(__dirname, '..', 'dist');
-if (config.output) {
-  outputDir = path.join(__dirname, '..', config.output);
-}
+const fm = require('json-front-matter');
+const marked = require('marked');
+const config = require(path.join(process.cwd(), 'config.json'));
+const source = path.join(process.cwd(), config.source || 'source');
+const outputDir = path.join(process.cwd(), config.output || 'public');
 
 function render (template, page, url) {
-  const output = (Handlebars.compile(template))(page);
-  if (!fs.existsSync(outputDir)) { fs.mkdirSync(outputDir); }
+  template = `${template}.hbs`;
+  const output = Handlebars.compile(
+    fs.readFileSync(path.join(process.cwd(), 'templates', template), 'utf-8')
+  )(page);
+  if (!fs.existsSync(outputDir)) {
+    fs.mkdirSync(outputDir);
+  }
   let dir = outputDir;
-  if (url) {
+  if (url !== 'index') {
     dir = path.join(dir, url);
-    if (!fs.existsSync(dir)){
+    if (!fs.existsSync(dir)) {
       fs.mkdirSync(dir);
     }
   }
-  fs.writeFile(path.join(dir, 'index.html'), output, 'utf8', (err) => { if (err) throw err; });
+  fs.writeFile(path.join(dir, 'index.html'), output, 'utf8', err => {
```

Comment:
i dont actually know
/me checks
the [docs](https://nodejs.org/api/fs.html#fs_fs_writefile_file_data_options_callback) say you should throw it